frontend/src/app/modules/auth/sign-up/sign-up.component.ts

```diff
@@ -0,0 +1,115 @@
+
+    confirmPassword: string = '';
+
+    submitted = false;
+
+    form: FormGroup = new FormGroup({
+        firstName: new FormControl(this.firstName),
+        lastName: new FormControl(this.lastName),
+        email: new FormControl(this.email),
+        password: new FormControl(this.password),
+        confirmPassword: new FormControl(this.confirmPassword),
+    });
+
+    constructor(private formBuilder: FormBuilder) {}
+
+    getEmailErrorMessage() {
+        if (this.form.controls['email'].hasError('required')) {
+            return 'You must enter a value';
+        }
+
+        if (this.form.controls['email'].hasError('email')) {
+            return 'Not a valid email';
+        }
+
+        return this.form.controls['email'].hasError('maxlength') ? 'Can not be more than 30 characters' : '';
+    }
+
+    getPasswordErrorMessage() {
+        if (this.form.controls['password'].hasError('required')) {
+            return 'Enter a value';
+        }
+
+        if (this.form.controls['password'].hasError('minlength')) {
+            return 'Can not be less than 2 symbols';
+        }
+
+        if (this.form.controls['password'].hasError('pattern')) {
+            return 'Should be at least one small and one capital letter';
+        }
+
+        return this.form.controls['password'].hasError('maxlength') ? 'Can not be more than 25 symbols' : '';
+    }
+
+    getConfirmPasswordErrorMessage() {
+        if (this.form.controls['confirmPassword'].hasError('required')) {
+            return 'Enter a value';
+        }
+
+        return this.confirmPassword !== this.password ? 'The password does not match' : '';
+    }
+
+    getNameErrorMessage(field: string) {
+        if (this.form.controls[field].hasError('required')) {
+            return 'Enter a value';
+        }
+
+        if (this.form.controls[field].hasError('minlength')) {
+            return 'Can not be less than 6 symbols';
+        }
+
+        return this.form.controls[field].hasError('maxlength') ? 'Can not be more than 25 symbols' : '';
+    }
+
+    validation() {
+        this.submitted = true;
+        this.validateData();
+    }
+
+    private validateData() {
+        this.form = this.formBuilder.group({
+            firstName: [this.firstName, [Validators.required, Validators.minLength(2), Validators.maxLength(25)]],
+            lastName: [this.lastName, [Validators.required, Validators.minLength(2), Validators.maxLength(25)]],
+            email: [this.email, [Validators.required, Validators.email, Validators.maxLength(30)]],
+            password: [
+                this.password,
+                [
+                    Validators.required,
+                    Validators.minLength(6),
+                    Validators.maxLength(25),
+                    Validators.pattern('^(?=.*[a-z])(?=.*[A-Z])[a-zA-Z\\d@$!%*?&\\.]{6,}$'),
+                ],
+            ],
+            confirmPassword: [this.confirmPassword, Validators.required],
+        });
```

Comment:
Can't you ad it onInit? Won't it work only after form is dirty and submitted?

Reply:
I tried this way at first, but it should accept the value and then validate, and not during initialization